hooks/tk-multi-publish2/maya/publish_lod_geometry.py

```
@@ -99,45 +101,39 @@ def settings_schema(self):
+        if not os.path.exists(publish_path_temp):
+            self.logger.error("Invalid Alembic path : {0}. Not renaming the top group.".format(
+                publish_path_temp))
             return
 
-        tmp_file = tempfile.NamedTemporaryFile(mode='w+b',
-                                               suffix='',
-                                               prefix='tmp',
-                                               dir=os.path.dirname(abc_file),
-                                               delete=True)
-
-        tmp_file_name = "{0}.abc".format(tmp_file.name)
-
-        # Loading the alembic archive in memory to process
-        abc_archive = cask.Archive(abc_file)
+        # Loading the alembic archive
+        abc_archive = cask.Archive(publish_path_temp)
 
         try:
             abc_archive.top.children.values()[0].name = rename_to
-            abc_archive.write_to_file(tmp_file_name, asOgawa=True)
-            # Renaming the temporary file to replace the original alembic
-            os.rename(tmp_file_name, abc_file)
-        except Exception as err:
-            self.logger.error("Unhandled Exceptions encountered. Securing original alembic.")
-            abc_archive.write_to_file(abc_file)
-            self.logger.error("File Secure Successful. Error Message below.")
-            self.logger.error(err.message)
+            abc_archive.write_to_file(publish_path, asOgawa=True)
+        except Exception:
+            import traceback
+            self.logger.error(
+                "Unhandled exceptions encountered.",
+                extra={
+                    "action_show_more_info": {
+                        "label": "Show Traceback",
+                        "tooltip": "Show complete traceback",
+                        "text": traceback.format_exc()
+                    }
+
+                })
+
+            self.logger.info("Attempting to recover original alembic file.")
+            self.logger.info("Recover Succesful.")
```

There is no recovery being done here...were you thinking of copying the temp file as the permanent file or something?
If not, you can remove these logs.
(Also, just saying, typo: successful)

---

Yeah, i was previously thinking that. But it makes no sense,to publish incorrect alembics. 
Forgot to get rid of those comments. Thanks

@typo : PyCharm's Fault :D

---

Okay, cool. Then maybe re-raise the exception? Otherwise publish will show up as successful.